### PR TITLE
new case:verify virtio-rng device work well on the OVMF guest

### PR DESCRIPTION
**Pull request by leidwang**
uefi_check_log_info.py:check virtio-rng device info in log

ID:1868268

Signed-off-by: Leidong Wang <leidwang@redhat.com>

### CODE REVIEW

**Comment by leidwang** [comment]
(1/2) Host_RHEL.m8.u3.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.uefi_check_rng_device.q35: PASS (53.83 s)
 (2/2) Host_RHEL.m8.u3.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.uefi_check_rng_device.q35: PASS (50.21 s)

**Comment by leidwang** [comment]
@XueqiangWei please help review it,many thanks!

File: qemu/tests/uefi_check_rng_device.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from virttest import utils_misc
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify virtio-rng device work well on the OVMF guest:
+    1) boot guest with virtio-rng device
+    3) check info in ovmf log
```
Comment (XueqiangWei) [inline qemu/tests/uefi_check_rng_device.py:12]: This case just check rng device info in ovmf log.

**Comment by XueqiangWei** [comment]
> uefi_check_rng_device.py:check virtio-rng device info in bios log
> 
> bz:1868268
> 
> Signed-off-by: Leidong Wang [leidwang@redhat.com](mailto:leidwang@redhat.com)

ID: 1868268, not bz: 1868268

File: qemu/tests/uefi_check_rng_device.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    def info_check(info):
+        """
+        Check log info
+        """
+        logs = vm.logsessions['seabios'].get_output()
+        result = re.search(info, logs, re.S)
+        return result
+
+    info_check_timeout = params.get("info_check_timeout", 360)
```
Comment (XueqiangWei) [inline qemu/tests/uefi_check_rng_device.py:27]: int()

Comment (leidwang) [reply qemu/tests/uefi_check_rng_device.py]: Yes,Thanks!

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by: xuwei@redhat.com

**Comment by leidwang** [comment]
@vivianQizhu please help review it,many thanks!

File: qemu/tests/cfg/uefi_check_rng_device.cfg [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    type = uefi_check_rng_device
+    only q35
+    only ovmf
+    kill_vm = yes
+    info_check_timeout = 360
+    virtio_rngs += " rng0"
+    backend_rng0 = rng-random
+    backend_type = passthrough
+    filename_passthrough = /dev/urandom
+    info_pattern = "InstallProtocolInterface\:\s3152BCA5-EADE-433D-862E-C01CDC291F44"
```
Comment (vivianQizhu) [inline qemu/tests/cfg/uefi_check_rng_device.cfg:11]: Is this code fixed on any host?

**Comment by vivianQizhu** [comment]
Do we have similar test case for seabios? Cause the case looks common for both of them.

File: qemu/tests/cfg/uefi_check_rng_device.cfg (hunk 3 above)
Comment (leidwang) [reply qemu/tests/cfg/uefi_check_rng_device.cfg:11]: Yes.

**Comment by leidwang** [comment]
> Do we have similar test case for seabios? Cause the case looks common for both of them.

There is no similar test case in seabios.

File: qemu/tests/uefi_check_rng_device.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    :param env: Dictionary with test environment.
+    """
+
+    def info_check(info):
+        """
+        Check log info
+        """
+        logs = vm.logsessions['seabios'].get_output()
+        result = re.search(info, logs, re.S)
+        return result
```
Comment (vivianQizhu) [inline qemu/tests/uefi_check_rng_device.py:25]: How about `return re.search(info, logs, re.S)`?

Comment (leidwang) [reply qemu/tests/uefi_check_rng_device.py:25]: Ok, it is more reasonable, thanks!

**Comment by vivianQizhu** [comment]
I think this one should share the same script with https://github.com/autotest/tp-qemu/pull/2332/files#diff-4969d3b89261e9a61adeb6b07e7ea1db and the script might be renamed. Or you abstract their common part into tp-qemu.provider.

**Comment by leidwang** [comment]
> I think this one should share the same script with https://github.com/autotest/tp-qemu/pull/2332/files#diff-4969d3b89261e9a61adeb6b07e7ea1db and the script might be renamed. Or you abstract their common part into tp-qemu.provider.

Ok,done,please help review it again, many thanks!

**Comment by vivianQizhu** [comment]
Please limit your commit title within 70 characters, otherwise LGTM.

**Comment by leidwang** [comment]
> Please limit your commit title within 70 characters, otherwise LGTM.

Ok,done.Thanks!

**Comment by leidwang** [comment]
Host_RHEL.m8.u3.product_av.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.uefi_check_log_info.rng_device.q35: PASS (60.95 s)

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**